src/component/calculator.js

```diff
@@ -1,52 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
+function Calculator() {
+  const [state, setState] = useState({ total: null, next: null, operation: null });
+
+  function handleClicks(e) {
+    setState((prevState) => ({ ...prevState, ...calculate(prevState, e.target.textContent) }));
   }
 
-  render() {
-    return (
-      <div className="mainContainer">
-        <div className="calculateDiv">
-          <div>
-            <input type="text" className="inputResult" placeholder="0" disabled />
-          </div>
-          <div className="pressValue">
-            <p>AC</p>
-            <p>+/-</p>
-            <p>%</p>
-            <p className="orange">+</p>
-          </div>
-          <div className="pressValue">
-            <p>7</p>
-            <p>8</p>
-            <p>9</p>
-            <p className="orange">X</p>
-          </div>
-          <div className="pressValue">
-            <p>4</p>
-            <p>5</p>
-            <p>6</p>
-            <p className="orange">-</p>
-          </div>
-          <div className="pressValue">
-            <p>1</p>
-            <p>2</p>
-            <p>3</p>
-            <p className="orange">+</p>
-          </div>
-          <div className="pressValue">
-            <p className="zero">0</p>
-            <p className="dot">.</p>
-            <p className="orange equalto">=</p>
+  return (
+    <div className="mainContainer">
+      <div className="calculateDiv">
+        <div>
+          <div type="text" className="inputResult">
+            {state.next || state.operation || state.total || 0}
           </div>
         </div>
+        <div className="pressValue">
+          <button type="button" onClick={handleClicks}>AC</button>
+          <button type="button" onClick={handleClicks}>+/-</button>
+          <button type="button" onClick={handleClicks}>%</button>
+          <button type="button" onClick={handleClicks} className="orange">+</button>
```

Comment:
- [ ] As far as I can see, you put another "+" button to your code rather than a "÷" button. Please try to implement the division on your calculator.

Reply:
operand mistake fixed